buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,352 @@
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+
+/**
+ *  Task to publish worker images to artifactory using buildkit
+ *  https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4063035406/BuildKit
+ */
+
+abstract class BuildkitBuild extends Exec {
+
+    private final String projectName = project.name
+    private final String version = project.version
+    private def gitTask
+    private def gitLogTask
+    private def releaseType
+
+    @Inject
+    protected abstract ObjectFactory getObjects()
+
+    @Inject
+    protected abstract ProviderFactory getProviderFactory()
+
+    // Credentials used for authentication to docker and artifactory
+    @Input
+    @Optional
+    final Property<String> registryUsername = getObjects().property(String).
+            convention(getProviderFactory().environmentVariable("CORDA_ARTIFACTORY_USERNAME")
+                    .orElse(getProviderFactory().gradleProperty("cordaArtifactoryUsername"))
+                    .orElse(getProviderFactory().systemProperty("corda.artifactory.username"))
+            )
+
+    @Input
+    @Optional
+    final Property<String> registryPassword = getObjects().property(String).
+            convention(getProviderFactory().environmentVariable("CORDA_ARTIFACTORY_PASSWORD")
+                    .orElse(getProviderFactory().gradleProperty("cordaArtifactoryPassword"))
+                    .orElse(getProviderFactory().systemProperty("corda.artifactory.password"))
+            )
+
+    // Handles to determind build type
+    @Input
+    final Property<Boolean> releaseCandidate =
+            getObjects().property(Boolean).convention(false)
+
+    @Input
+    final Property<Boolean> nightlyBuild =
+            getObjects().property(Boolean).convention(false)
+
+    @Input
+    final Property<Boolean> preTest =
+            getObjects().property(Boolean).convention(false)
+
+    // Handle to set the build tool used to create an image
+    // Currently supported tools are docker buildx and native builkit 
+    // NOTE: native buidkit requires port forwarding to the aws buildkit cluster
+    @Input
+    final Property<Boolean> isBuildx =
+            getObjects().property(Boolean).convention(true)
+
+    // Handle for loading images into docker
+    @Input
+    final Property<Boolean> useDocker =
+            getObjects().property(Boolean).convention(false)
+
+    @Input
+    final Property<Boolean> useShortName =
+            getObjects().property(Boolean).convention(false)
+
+    // Handle to set the output image's repository:name:tag
+    @Input 
+    final Property<String> containerRepo =
+            getObjects().property(String).convention('')
+
+    @Input
+    final Property<String> containerTag =
+            getObjects().property(String).convention('')
+
+    @Input
+    final Property<String> containerName =
+            getObjects().property(String).convention('')
+
+    // Handles to set the base image's name:tag
```

Comment:
Can you replace the word handle with property here in all instances , there's a few